### ISSUE TITLE
any kind of sharable documentation on the fields and meanings and such?

### ISSUE THREAD

**Issue by daym** [oxidecomputer/amd-apcb#98]
any kind of sharable documentation on the fields and meanings and such?

**Comment by daym** [comment]
Yeah, we will add some more documentation on what the actual settings mean.

**Comment by hanetzer** [comment]
nice, yeah. As much as I appreciate source, I'm not a native rust speaker, so making heads and tails of it is tricky for me.

**Comment by daym** [comment]
You can make it dump the schema via this program:

```
use amd_apcb::Apcb;
use schemars::schema_for;

fn main() {
    let schema = schema_for!(Apcb);
    println!("{}", serde_json::to_string_pretty(&schema).unwrap());
}
```

The result is a json schema (in json). You can use any JSON schema aware editor to edit your config files. I use IntelliJ IDEA. You can associate the schema in the editor at the bottom statusbar.

There's a program to generate the schema out of the box (and the entire flash editing), but it's in review.
